### PR TITLE
Births

### PR DESCRIPTION
**Pull request by robynstuart**
@cliffckerr please review.

Implements births, MTCT, PMTCT, age transitions, risk transitions. 

Still to do: circs, check that everything makes sense, check that PMTCT has appropriate effects.

### CODE REVIEW

File: optima/model.py [hunk 1]
```diff
@@ -105,6 +107,8 @@ def model(simpars=None, settings=None, verbose=2, safetymargin=0.8, benchmark=Fa
     circum   = simpars['circum']    # Prevalence of circumcision (P)
     stiprev  = simpars['stiprev']   # Prevalence of STIs (P)
     prep     = simpars['prep']      # Prevalence of PrEP (P)
+    numpmtct = simpars['numpmtct']  # Number (or proportion?) of people receiving PMTCT (P/N)
+    usepmtctprop=True if all(numpmtct<1) else False
```
Comment (cliffckerr) [inline optima/model.py]: this reminds me of what we had in optima 1.0 which...didn't work :(

File: optima/parameters.py [hunk 2]
```diff
@@ -344,7 +360,7 @@ def makepars(data, label=None, verbose=2):
     # Set up keys
     totkey = ['tot'] # Define a key for when not separated by population
     popkeys = data['pops']['short'] # Convert to a normal string and to lower case...maybe not necessary
-    fpopkeys = [popkeys[i] for i in range(len(popkeys)) if pars['female'][i]]
+    fpopkeys = [popkey for popno,popkey in enumerate(popkeys) if data['pops']['female'][popno]]
```
Comment (cliffckerr) [inline optima/parameters.py]: sorry, what's the reason for this change?

File: optima/parameters.py [hunk 3]
```diff
@@ -386,26 +402,43 @@ def makepars(data, label=None, verbose=2):
     ## Tidy up -- things that can't be converted automatically
     ###############################################################################    
     
-    # Births
+    # Births rates. This parameter is coupled with the birth matrix defined below
     for key in list(set(popkeys)-set(fpopkeys)): # Births are only female: add zeros
         pars['birth'].y[key] = array([0])
         pars['birth'].t[key] = array([0])
+    pars['birth'].y = pars['birth'].y.sort(popkeys) # Sort them so they have the same order as everything else
+    pars['birth'].t = pars['birth'].t.sort(popkeys)
```
Comment (cliffckerr) [inline optima/parameters.py]: ah, good point -- this probably needs to be done for circumcision etc. as well

File: optima/parameters.py [hunk 4]
```diff
@@ -462,7 +496,8 @@ def makesimpars(pars, inds=None, keys=None, start=2000, end=2030, dt=0.2, tvec=N
     
     # Copy default keys by default
     for key in generalkeys: simpars[key] = dcp(pars[key])
-    
+    for key in staticmatrixkeys: simpars[key] = array(pars[key])
```
Comment (cliffckerr) [inline optima/parameters.py]: should probably dcp, no?

File: tests/testmakespreadsheet.py [hunk 5]
```diff
@@ -9,7 +9,7 @@
 ## Define tests to run here!!!
 tests = [
 'makespreadsheet',
-#'unittests',
+'unittests',
```
Comment (cliffckerr) [inline tests/testmakespreadsheet.py]: the problem with the unittests is that they kill the console (at least for me...?)

File: optima/model.py (hunk 1 above)
Comment (robynstuart) [inline optima/model.py]: no, what didn't work in v1 was that we checked the entries separaterly on each time loop, like
`if numpmtct[t]>1` for each t.
The broke because it could have been 0 at one time step but a number the next. I think it'd valid to do it this way, however, where you check the whole array at once.

File: optima/parameters.py (hunk 2 above)
Comment (robynstuart) [inline optima/parameters.py]: I wanted to see if it was the same thing :) actually I did change it back, but i see that I forgot to push.

File: tests/testmakespreadsheet.py (hunk 5 above)
Comment (robynstuart) [inline tests/testmakespreadsheet.py]: not for me. all the GUIs do however!!!

File: optima/model.py (hunk 1 above)
Comment (cliffckerr) [inline optima/model.py]: it still feels very fragile and hacky to me; i'd prefer to have separate parameters, but that has its pitfalls too. i don't have a better suggestion so i'm ok to leave it as is for now.
